File: 프로그래머스/Lv.3/43162. 네트워크/네트워크.py
def dfs(computers, visited, node):
    visited[node]=True
    for idx, value in enumerate(computers[node]):
        if idx!=node and value==1 and not(visited[idx]):
            dfs(computers, visited,idx)
            #print(node, idx, child)



def solution(n, computers):
    visited = [False] * n
    answer = 0
    start = 0
    visited[start] = start
    for node in range(n):
        if not(visited[node]):
            dfs(computers, visited, node)
            answer+=1

    return answer

File: JS/dfs/네트워크.js
function dfs(visited, computers, curr) {
  visited[curr] = true;
  for (let idx = 0; idx < computers[curr].length; idx++) {
    // console.log(curr, idx, computers[curr][idx])
    if (idx != curr && computers[curr][idx] == 1 && visited[idx] == false) {
      dfs(visited, computers, idx);
    }
  }
}

function solution(n, computers) {
  var answer = 0;
  let visited = Array.from({ length: n }, () => false);
  for (let curr = 0; curr < n; curr++) {
    if (visited[curr] == false) {
      dfs(visited, computers, curr);
      answer++;
    }
  }
  return answer;
}
